clap_complete/src/shells/bash.rs

```diff
@@ -14,7 +14,9 @@ impl Generator for Bash {
     }
 
     fn generate(&self, app: &App, buf: &mut dyn Write) {
-        let bin_name = app.get_bin_name().unwrap();
+        let bin_name = app
+            .get_bin_name()
+            .expect("Getting the App's \"bin_name\" failed");
```

Comment:
This reads like its describing `app.get_bin_name().unwrap()` and isn't adding any value.

A tip for writing `expect`s is to use them to document what the expected pre-condition is that failed.  In this case, `crate::generate set the bin_name `

Reply:
> and isn't adding any value.

i mean its still better than `unwrap failed at`, this way you currently have at least the clue what property is missing before having to investigate the code that threw the message

but yes, the message is not really great

how about changing the message to:
`Getting the App's \"bin_name\" failed: did crate::generate set the \"bin_name\" correctly?`

Reply:
The distinction is expects are not error reporting but a form of assertion.  Developers get a traceback telling them where it failed and can easily see what the line is that failed.  The emphasis should be on the documentating the invariant (by emphasis, I mean linguistically)

This goes back to the tip I mentioned.  A more concrete way of putting it is to turn "expect("...")` into "expect  ..." and fill in the sentence fragment.  In this case "expect crate::generate set the bin_name" which becomes `expect("crate::generate set the bin_name")`.  Yes, the panic message isn't optimized for this (its a common complaint of expect's panic message).

Reply:
> expect crate::generate set the bin_name
> crate::generate set the bin_name

they dont sound descriptive to me, how about `Expected crate::generate to have set "bin_name"` and maybe with the addition of `(Did this function get run directly?)` as a hint to not run this function directly (but through `crate::generate`)